File: 13.py
class Solution:
    valDict: dict[str, int] = {
        'I': 1,
        'V': 5,
        'X': 10,
        'L': 50,
        'C': 100,
        'D': 500,
        'M': 1000
    }
    def romanToInt(self, s: str) -> int:
        totalVal = 0
        lastVal = 1001
        for c in s:
            val = self.valDict[c]
            if val > lastVal:
                totalVal -= 2 * lastVal
            totalVal += val
            lastVal = val
        return totalVal

File: sol/13.cpp
struct Solution {
    int romanToInt(const string &s) {
        int totalVal = 0, lastVal = 1001, val;
        for (char c: s){
            switch (c){
                case 'I':
                    val = 1;
                    break;
                case 'V':
                    val = 5;
                    break;
                case 'X':
                    val = 10;
                    break;
                case 'L':
                    val = 50;
                    break;
                case 'C':
                    val = 100;
                    break;
                case 'D':
                    val = 500;
                    break;
                case 'M':
                    val = 1000;
                    break;
            }
            if (val > lastVal){
                totalVal -= lastVal << 1;
            }
            totalVal += val;
            lastVal = val;
        }
        return totalVal;
    }
};
